### PR TITLE
[#40866] .github: workflows: farshow-build.yml: Add actions script

### PR DESCRIPTION
**Pull request by ivysochyn**
Adds script for GitHub Actions performing library lint checking and building an 'example-build'.

### CODE REVIEW

File: .github/workflows/farshow-build.yml [hunk 1]
```diff
@@ -0,0 +1,98 @@
+name: Farshow build
+
+on: [push]
```
Comment (glatosinski) [inline .github/workflows/farshow-build.yml]: I guess we should:
```
on:
  push:
    branches: [main]
  pull_request:
    branches: [main]
```

File: .github/workflows/farshow-build.yml [hunk 2]
```diff
@@ -0,0 +1,98 @@
+name: Farshow build
```
Comment (glatosinski) [inline .github/workflows/farshow-build.yml]: Lower case for farshow

**Review by glatosinski — CHANGES_REQUESTED** [review]
In general LGTM, just some minor things to fix

File: .github/workflows/farshow-build.yml [hunk 3]
```diff
@@ -0,0 +1,98 @@
...
+          cmake \
+          g++ \
+          git \
+          libglew-dev \
+          libglfw3-dev \
+          libopencv-dev \
+          libopengl-dev \
+          make
+
+    - name: Add repository to safe list
```
Comment (glatosinski) [inline .github/workflows/farshow-build.yml:30]: Why do we need that?

File: .github/workflows/farshow-build.yml [hunk 4]
```diff
@@ -0,0 +1,98 @@
...
+      with:
+        submodules: recursive
+
+    - name: Initialize building environment
+      run: |
+        mkdir build
+        cd build
+        cmake -DCMAKE_EXPORT_COMPILE_COMMANDS=ON ..
+
+    - name: Format
```
Comment (glatosinski) [inline .github/workflows/farshow-build.yml]: I guess this and next step could be wrapped in one task called `Code linting` or something like that

File: .github/workflows/farshow-build.yml [hunk 5]
```diff
@@ -0,0 +1,98 @@
...
+
+    - name: Add repository to safe list
+      run: git config --global --add safe.directory "$GITHUB_WORKSPACE"
+
+    - name: Checkout sources
+      uses: actions/checkout@v3
+      with:
+        submodules: recursive
+
+    - name: Initialize building environment
```
Comment (glatosinski) [inline .github/workflows/farshow-build.yml]: I'd call it `Build the project` and add building the project before linting

File: .github/workflows/farshow-build.yml [hunk 6]
```diff
@@ -0,0 +1,98 @@
...
+      run: |
+        cd build
+        make format
+
+    - name: Test if changes were applied
+      run: test -z $(git diff) && exit 0 || exit 1
+
+  export-test:
+    runs-on: ubuntu-latest
+    needs: lint
```
Comment (glatosinski) [inline .github/workflows/farshow-build.yml]: I don't think it needs `lint` - we can build it anyway, it's independent check that is also informative

File: .github/workflows/farshow-build.yml [hunk 7]
```diff
@@ -0,0 +1,98 @@
...
+          libopencv-dev \
+          libopengl-dev \
+          make
+
+    - name: Checkout sources
+      uses: actions/checkout@v3
+      with:
+        submodules: recursive
+
+    - name: Initialize building environment
```
Comment (glatosinski) [inline .github/workflows/farshow-build.yml]: Maybe just `Build the project`, and later `Install the project`.

**Review by glatosinski — CHANGES_REQUESTED** [review]
Added some more comments

File: .github/workflows/farshow-build.yml (hunk 3 above)
Comment (ivysochyn) [reply .github/workflows/farshow-build.yml:30]: The `git` command won't be initialized unless the repository directory is safe.

You can see this behavior in the "Test if changes were applied" task for the next pipeline:
https://github.com/antmicro/farshow/actions/runs/3957311195/jobs/6777523261

**Review by glatosinski — APPROVED** [review]
LGTM

**Pull request merged**